File: agent/src/main/java/eu/cqse/teamscale/jacoco/agent/AgentOptionParseException.java
package eu.cqse.teamscale.jacoco.agent;

/**
 * Thrown if option parsing fails.
 */
public class AgentOptionParseException extends Exception {

	/**
	 * Serialization ID.
	 */
	private static final long serialVersionUID = 1L;

	/**
	 * Constructor.
	 */
	public AgentOptionParseException(String message) {
		super(message);
	}

	/**
	 * Constructor.
	 */
	public AgentOptionParseException(String message, Throwable cause) {
		super(message, cause);
	}

}

File: agent/src/main/java/com/teamscale/jacoco/agent/AgentOptionParseException.kt
package com.teamscale.jacoco.agent

/**
 * Thrown if option parsing fails.
 */
class AgentOptionParseException : Exception {

    /**
     * Constructor.
     */
    constructor(message: String) : super(message) {}

    /**
     * Constructor.
     */
    constructor(message: String, cause: Throwable) : super(message, cause) {}

    companion object {

        /**
         * Serialization ID.
         */
        private val serialVersionUID = 1L
    }

}
